jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/S3Fields.kt

```diff
@@ -0,0 +1,57 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package software.aws.toolkits.jetbrains.services.s3
+
+import software.amazon.awssdk.services.s3.S3Client
+import java.time.Instant
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Request
+
+// S3 Key class represents a base class for S3 Directory and S3 Objects
+
+sealed class S3Key(val bucket: String, val key: String) {
+    open val name = if (key.endsWith("/")) {
+        key.dropLast(1)
+    } else {
+        key.substringAfterLast("/")
+    }
+}
+
+class S3Bucket(
+    override val bucketName: String,
```

Comment:
is there a difference between `bucketName` and `bucket` that's defined on the parent class (`S3Key`)? I don't see the need to `override` the value in each implementation.